util/new_project.sh

```diff
@@ -24,20 +24,36 @@ if [ -e "keyboards/$1" ]; then
 	exit 1
 fi
 
-cd "$(dirname "$0")/.."
+cd "$(dirname "$0")/.." || exit
 
-KEYBOARD_UPPERCASE=$(echo $1 | awk '{print toupper($0)}')
-KEYBOARD_NAME=$(basename $1)
-KEYBOARD_NAME_UPPERCASE=$(echo $KEYBOARD_NAME | awk '{print toupper($0)}')
+KEYBOARD_NAME=$(basename "$1")
+KEYBOARD_NAME_UPPERCASE=$(echo "$KEYBOARD_NAME" | awk '{print toupper($0)}')
 
 
-cp -r quantum/template/base keyboards/$KEYBOARD
-cp -r quantum/template/$KEYBOARD_TYPE/. keyboards/$KEYBOARD
+cp -r quantum/template/base "keyboards/$KEYBOARD"
+cp -r "quantum/template/$KEYBOARD_TYPE/." "keyboards/$KEYBOARD"
 
-mv keyboards/${KEYBOARD}/template.c keyboards/${KEYBOARD}/${KEYBOARD_NAME}.c
-mv keyboards/${KEYBOARD}/template.h keyboards/${KEYBOARD}/${KEYBOARD_NAME}.h
-find keyboards/${KEYBOARD} -type f -exec sed -i '' -e "s;%KEYBOARD%;${KEYBOARD_NAME};g" {} \;
-find keyboards/${KEYBOARD} -type f -exec sed -i '' -e "s;%KEYBOARD_UPPERCASE%;${KEYBOARD_NAME_UPPERCASE};g" {} \;
+mv "keyboards/${KEYBOARD}/template.c" "keyboards/${KEYBOARD}/${KEYBOARD_NAME}.c"
+mv "keyboards/${KEYBOARD}/template.h" "keyboards/${KEYBOARD}/${KEYBOARD_NAME}.h"
+find "keyboards/${KEYBOARD}" -type f -exec sed -i '' -e "s;%KEYBOARD%;${KEYBOARD_NAME};g" {} \;
+find "keyboards/${KEYBOARD}" -type f -exec sed -i '' -e "s;%KEYBOARD_UPPERCASE%;${KEYBOARD_NAME_UPPERCASE};g" {} \;
+
+GIT=$(whereis git)
+if [ "$GIT" != "" ]; then
```

Comment:
Disagree on `-n`, you have to look that up ln the horrible shell manuals. It's idiomatic shell to compare against empty string. One uses a variable alias so you can update them all at once, otherwise you have to go edit every single git command. Been there done that got the t-shirt. e.g. let's say we wanted to change it to `hub`.

Reply:
also disagree with how you invoke the command in the if statement. That's a shellshock warning right there. It's also not clear what you are checking for. I'd rather be explicit than implicit for something as wonky as shell.

Reply:
Wrote that on mobile so it added a line break where it shouldn't have; fixed now.

I think `--is-inside-work-tree` is pretty clear, and for what it is it's more explicit than calling `git log` and checking the exit code. Doing that begs the question: why `log` and not some other command when it appears to not make a difference? With `rev-parse --is-inside-work-tree` the command itself states the intent, and it's intended for this use case.

If you don't think it's explicit enough, you can always put it in a variable or add a comment:
```sh
is_git_repo=$(git rev-parse --is-inside-work-tree 2>/dev/null)
if [ "$is_git_repo" = "true" ]; then
  id="'$(git config --get user.name)'"
  # ...
fi
```
(Note the lowercase variable names, since they're not constants or env vars.)

Re `$GIT`: you'd still have to go through every file to replace it, so I don't see how this makes it any better for the two lines it's used on here, especially since no other script does it that way.

Also, you mean ShellCheck, not ShellShock, right?

Reply:
The style the file uses is CAPITALIZED variables. That's actually a thing in shell to differentiate variables. It was that way before I edited the file, check the log please. You are correct shell check. Not shock.

I think the test you propose is fine, but the variable naming makes it clearer what the intent is already, and is functionally equivalent with less code. KISS is a thing.

I agree with the ID optimization, it used to have email in there, I removed it because of privacy concerns. Thanks for your suggestions, but I think you're applying different standards to file rather than matching the style that was already there. Variables for binaries is a pretty normal shell practice, so I think we'll just have to disagree on this.

The

Reply:
Hey, so `rev-parse` doesn't actually work the way you think it does. It simply errors out when not in a git repo. It only returns true or false inside an actual git repo, for conditions that we're not interested in.

```console
➜  ~ git rev-parse --is-inside-git-dir
fatal: not a git repository (or any of the parent directories): .git
➜  ~ echo $?
128
```

As it basically has exactly the same behavior I don't really see any advantage to using this command and flag in this particular context. I will update the ID code though, as that is a good suggestion as we're no longer using email.

Reply:
does exactly the same thing:

```console
➜  projects git rev-parse --is-inside-work-tree
fatal: not a git repository (or any of the parent directories): .git
```

Sorry, I think checking return codes are better than strings in shell, so I'll leave it as is. Your code also doesn't work if git isn't installed, which could happen if someone downloads the source, so I'm going to leave it as is. Thanks for the feedback.

Reply:
As intended; again, that's what the `2>/dev/null` is for.
Alright, sounds good. You're welcome!